src/Runner.Worker/ExecutionContext.cs

```diff
@@ -995,8 +1041,7 @@ public void PublishStepTelemetry()
                         StepTelemetry.FinishTime = _record.FinishTime;
                     }
 
-                    if (!IsEmbedded &&
-                        _record.Issues.Count > 0)
```

Comment:
the only thing going on in this conditional is a `foreach`.  `foreach` is a no-op for an empty list.

Reply:
i would still keep it to avoid waste CPU.